syntax = "proto3";

package proto;

message DictionaryData {
    map<string, string> data = 1;
    int32 universal_id = 2; 
}

message TextData {
    string data = 1;
    int32 universal_id = 2; 
}

message SpeechData {
    string data = 1;
    int32 universal_id = 2; 
}


message Ack {
    bool success = 1;
    string message = 2;
    int32 universal_id = 3; 
}


service ModuleB {
    rpc eventToTextRequest(DictionaryData) returns (Ack);
}

service ModuleC {
    rpc textToSpeechRequest(TextData) returns (Ack);
}

service ModuleD {
    rpc ReproduceSpeechRequest(SpeechData) returns (Ack);
}